docs/refs/api/drivers/InterruptIn.md

```diff
@@ -12,6 +12,8 @@ Use the InterruptIn interface to trigger an event when a [digital input pin](Dig
 
 * No printf, malloc or new in ISR: avoid any call to bulky library functions. In particular, certain library functions (such as printf, malloc and new) are non re-entrant, and their behavior could be corrupted when called from an ISR.
 
+* For printf's from interrupt context use [Events](../rtos/events.md) instead. 
```

Comment:
Query: What is this sentence trying to say? I don't understand it.

Reply:
the previous bullet point says 'dont do printf from interrupt context', this bullet point has the suggested fix